thoth/adviser/resolver.py

```diff
@@ -1302,18 +1303,21 @@ def resolve_products(
 
     def resolve(self, *, with_devel: bool = True, user_stack_scoring: bool = True) -> Report:
         """Resolve software stacks and return resolver report."""
-        report = Report(count=self.count, pipeline=self.pipeline)
+        states: List[Tuple[Tuple[float, int], State]] = []
+        heapq_counter = 0  # Making sure the first resolved state takes precedence when adding to the report.
 
         self._init_context()
         with Unit.assigned_context(self.context), self.predictor.assigned_context(self.context):
             for state in self._do_resolve_states(with_devel=with_devel, user_stack_scoring=user_stack_scoring):
-                if report.will_add_product_from_state(state.score):
-                    # An optimization - we do not need to run wraps if the product is not part of the final result.
-                    self._run_wraps(state)
-                    product = Product.from_final_state(context=self.context, state=state)
-                    report.add_product(product)
+                item = ((state.score, heapq_counter), state)
```

Comment:
Can you add a small comment that item[0] is being used only for sorting?

Reply:
Adjusted in c6dade99. 👍🏻